client/src/components/BuySell.jsx

```diff
@@ -0,0 +1,1341 @@
+      super(props);
+      this.state = {
+        stockName: '',
+        stockTicker: '',
+        currentPrice: 0,
+        shares: 0,
+        estimatedCost: 0,
+        orderDescription: 'Estimated Cost',
+        userShares: 103,
+        userBalance: 246.42,
+        sharesBalance: ``,
+        borderTopColor: '',
+        borderBottomColor: '',
+        borderLeftColor: '',
+        borderRightColor: '',
+        clickBorderBottomColor: '',
+        color: '',
+        marginInlineEnd: '',
+        marginRight: '',
+        width: '',
+        isSelectedBuy: true,
+        isSelectedSell: false
+      }
+      this.buyStock = this.buyStock.bind(this);
+      this.handleChange = this.handleChange.bind(this);
+      this.buySellClick = this.buySellClick.bind(this);
+      this.sharesClick = this.sharesClick.bind(this);
+      this.sharesBlur = this.sharesBlur.bind(this);
+      this.sharesHover = this.sharesHover.bind(this);
+      this.sharesLeaveHover = this.sharesLeaveHover.bind(this);
+      this.changeStock = this.changeStock.bind(this);
+  }
+
+  componentWillMount() {
+    this.changeStock();
+  }
+
+  componentDidMount() {
+    this.setState({
+      sharesBalance: `${this.state.userBalance} Buying Power Available`,
+      isSelectedBuy: true,
+      isSelectedSell: false
+    })
+  }
+
+  changeStock() {
+    axios.get('/stocks').then(res => {
+      const stocks = res.data;
+      const id = parseInt(window.location.pathname.slice(1, window.location.pathname.length-1))
+      var urlStock = {}
+      for (var i=0; i < stocks.length; i++) {
+        if (stocks[i]._id === id) {
+            urlStock = stocks[i];
+        }
+      }
+      console.log('STOCCCCCCKK', urlStock, id)
+      this.setState({
+        stockName: urlStock.stockName,
+        stockTicker: urlStock.stockTicker,
+        currentPrice: urlStock.currentPrice,
+      })
+    })
+  }
+
+  sharesHover() {
+    if (this.state.borderBottomColor !== 'rgb(33, 206, 153)'){
+      this.setState({
+        borderTopColor: 'rgb(132, 132, 134)',
+        borderBottomColor: 'rgb(132, 132, 134)',
+        borderLeftColor: 'rgb(132, 132, 134)',
+        borderRightColor: 'rgb(132, 132, 134)'
+      })
+    }
+  }
+
+  sharesLeaveHover() {
+    if (this.state.borderBottomColor !== 'rgb(33, 206, 153)'){
+      this.setState({
+        borderTopColor: 'rgb(23, 23, 24)',
+        borderBottomColor: 'rgb(23, 23, 24)',
+        borderLeftColor: 'rgb(23, 23, 24)',
+        borderRightColor: 'rgb(23, 23, 24)'
+      })
+    }
+  }
+
+  sharesClick(e) {
+    this.setState({
+      borderTopColor: 'rgb(33, 206, 153)',
+      borderBottomColor: 'rgb(33, 206, 153)',
+      borderLeftColor: 'rgb(33, 206, 153)',
+      borderRightColor: 'rgb(33, 206, 153)'
+    })
+  }
+
+  sharesBlur() {
+    this.setState({
+      borderTopColor: 'rgb(23, 23, 24)',
+      borderBottomColor: 'rgb(23, 23, 24)',
+      borderLeftColor: 'rgb(23, 23, 24)',
+      borderRightColor: 'rgb(23, 23, 24)'
```

Comment:
seems like these strings are being repeated an awful lot.. can we just store these strings elsewhere and instead set state to strings like "dark gray", "light gray", "yellow" (which then reference these specific RGB values)?